website_logo/controllers/main.py

```diff
@@ -0,0 +1,68 @@
+
+import functools
+import logging
+import io
+import base64
+
+from odoo import _, http, SUPERUSER_ID
+from odoo.http import request
+from odoo.modules import get_module_resource, registry
+
+db_monodb = http.db_monodb
+_logger = logging.getLogger(__name__)
+
+
+class Website(http.Controller):
+
+    def _image_logo_get(self, cr, domain=None):
+        if domain:
+            cr.execute("""SELECT logo, write_date
+                            FROM website
+                           WHERE domain = %s
+                       """, (domain,))
+        else:
+            cr.execute("""SELECT logo, write_date
+                            FROM website
+                       """)
+        row = cr.fetchone()
+        if row and row[0]:
+            return io.BytesIO(base64.b64decode(row[0])), row[1]
+        return False, False
+
+    @http.route([
+        '/web/binary/website_logo',
+        '/website_logo',
+        '/website_logo.png',
+    ], type='http', auth="none", cors="*")
+    def website_logo(self, dbname=None, **kw):
+        imgname = 'website_nologo.png'
```

Comment:
nologo image seems empty, am I wrong? Should we have a `no logo` image or the default odoo one?

Reply:
It is the text placed in Usage section of README.rst, it was written before the migration:

This addon supports multi-website. If no logo is defined, a transparent image is
placed, allowing to have no logo in one website, even if "Show logo" option is
set.